### DISCUSSION TITLE
Are PS1 or CMD commands recorded?

### DISCUSSION THREAD

**Discussion by innovate-support** [Security-Onion-Solutions/securityonion#5968] [Unsupported Versions] [answered]
I'm spending quite a bit of time Hunting for things and finding amazing results, but I'm wondering if SO/Wazuh records the actual commands that are run in Powershell or CMD terminals?  I can Hunt for and see a cmd.exe processes ran, but I'm not seeing what command was actually entered.  Is there a way to track and view that?

**Comment by dougburks** [comment]
Do you have fields like `process.command_line` and `process.parent.command_line`?
![Screen Shot 2021-10-22 at 3 47 20 PM](https://user-images.githubusercontent.com/1659467/138514494-cbd642c1-5159-4261-adc6-709ce3fb581d.png)

**Reply by innovate-support** [reply]
no, I don't have those fields. 
![Screenshot from 2021-10-22 17-11-03](https://user-images.githubusercontent.com/20424476/138532246-db61e280-96cb-4d94-920a-86648213abb9.png)

**Reply by dougburks** [reply]
Do you have sysmon installed on the endpoint?

**Comment by innovate-support** [comment, marked as answer]
Nope, I wasn't aware **Sysmon** was a requirement.  So now I have it working.  Here's what I did on the Endpoint.

1. I installed Sysmon64 using it's default configuration.
[Download Sysmon](https://docs.microsoft.com/en-us/sysinternals/downloads/sysmon)

2. I edited the **ossec.conf** and added this.
```
<localfile>
   <location>Microsoft-Windows-Sysmon/Operational</location>
   <log_format>eventchannel</log_format>
</localfile>
```

3. I restarted the Wazuh Agent via the Wazuh Manage Agent gui tool

4. I opened both a CMD terminal and PS terminal and did a simple ping test 
`ping 1.1.1.1`

5. I opened up the SO Server web gui and Hunted using the following command.  My endpoint agent name is Hawkeye.  
`agent.name: "Hawkeye" | groupby process.command_line @timestamp`
